base/test.jl

```diff
@@ -908,4 +908,11 @@ function detect_ambiguities(mods...; imported::Bool=false)
     collect(ambs)
 end
 
+immutable GenericString <: AbstractString
+    string::AbstractString
```

Comment:
If this is intended to be used by packages, it'd be good to provide a docstring, because I imagine the appearance of this type in this file would appear strange to the casual - doesn't even necessarily have to be in the manual, but that'd be nice too.